### ISSUE TITLE
[Conduit] Does not work as of newest release (0.5.0)

### ISSUE THREAD

**Issue by jfowl** [Uberspace/lab#1460]
TL;DR: Pre-build binaries are glibc-incompatible with uberspace, guide broken.

Previously, Conduit was available as a statically linked binary, so it also worked here. Due to static compilation ~~misery~~ complications, I dropped that and switched to just linking to the default glibc of Debian bullseye, which is a tad newer than CentOS 7's glibc. As a result, 0.5 prebuilt binaries do not work on Uberspace anymore. As a result, the download URLs changed, so following the guide's first step results in a 404.

Building conduit on Uberspace with `cargo build` also does not work, due to missing build dependencies (`libclang-dev`, on debian).

So until further notice, this guide is broken, and new conduit versions can't be installed.
The latest working version can be retrieved at https://gitlab.com/famedly/conduit/-/package_files/43706908/download.

I'll try to remember to update this if I get around to build Conduit with an old enough glibc version to work here or if static builds are possible again.

**Comment by jfowl** [comment]
> Is there any news here? It would be great to be able to bring Conduit up to date.

Sorry, no capacity right now. I remember seeing some success at static-building with https://github.com/rust-cross/cargo-zigbuild in the conduit chat though: https://matrix.to/#/!SMloEYlhCiqKwRLAgY:fachschaften.org/$5kog8JFrjMv1W6Pt4jNd_TKfqIMprTAaVMS05ifpRCA?via=flipdot.org&via=matrix.org&via=envs.net

So maybe this is a possibility for you to take a dig at it

**Comment by 23n** [comment]
Version 0.6.0 is out, do we have a chance to get it running again on an Uberspace? I'm still stuck on v0.4.0.

**Comment by jfowl** [comment]
So I was nearly able to compile it on `wild.uberspace.de` with rocksdb disabled:

```bash
git clone https://gitlab.com/famedly/conduit.git /tmp/conduit
cd /tmp/conduit
cargo build --no-default-features --features "conduit_bin,backend_sqlite"
```

But it got killed with signal 9, so I assume uberspace's resource restrictions do not allow such CPU hungry compile times.

So maybe this is a way to go forward :)

**Comment by coderkun** [comment]
In case anyone cares, [Dendrite](https://matrix-org.github.io/dendrite/) works well on Uberspace. If I find some time, I will submit a guide for it.

**Comment by samuelhaeussler** [comment]
> In case anyone cares, [Dendrite](https://matrix-org.github.io/dendrite/) works well on Uberspace. If I find some time, I will submit a guide for it.

I tried myself setting up dendrite, but couldn't get it running. Do you have some documentation for it?

**Comment by coderkun** [comment]
> > In case anyone cares, [Dendrite](https://matrix-org.github.io/dendrite/) works well on Uberspace. If I find some time, I will submit a guide for it.
> 
> I tried myself setting up dendrite, but couldn't get it running. Do you have some documentation for it?

Not yet but I created #1712 for it.

**Comment by iiiw** [comment]
conduit (still/again?) provides static binaries here:
https://docs.conduit.rs/deploying/generic.html#installing-conduit 

I guess the only change that is necessary to the guide is to use the new URL here:
`wget https://gitlab.com/api/v4/projects/famedly%2Fconduit/jobs/artifacts/master/raw/x86_64-unknown-linux-musl?job=artifacts -O ~/bin/conduit`

Updating to v0.10.1 with this link seems to work fine.
